Yank/multistate/multistatesampler.py

```diff
@@ -1140,14 +1188,18 @@ def _compute_energies(self):
     def _compute_replica_energies(self, replica_id):
         """Compute the energy for the replica in every ThermodynamicState."""
         # Initialize replica energies for each thermodynamic state.
-        energy_thermodynamic_states = np.zeros(self.n_states)
+        energy_thermodynamic_states = np.ma.zeros(self.n_states)
```

Comment:
You said you got rid of all masks, but there is one here. Since it was causing some undefined behavior in your tests according to you, then I don't think you intended for this to still be here. Probably best to remove it so it does not cause other unforeseen issues

Reply:
Good catch!